client/src/panel.tsx

```diff
@@ -288,6 +288,10 @@ export const Panel: React.FC<IsDropTargetProps> = ({ isDropTarget }) => {
           <Pinboard
             key={pinboardId}
             pinboardId={pinboardId}
+            composerId={
+              activePinboards.find((_) => _.id === pinboardId)?.composerId ||
```

Comment:
Definitely non-blocking – could this be a bit slow with large numbers of pinboards, or is that unlikely? Conscious that this will run every render IIUC, and unsure how often that will happen given pinboard's a real time tool – could e.g. memoise tuple of active pinboard ids and composer ids if that was a concern.

Reply:
good spot, memoised that now